scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -1120,4 +1124,42 @@ public void removeThirdPartyCredential(String key) throws NotConnectedException,
                 "You do not have permission to remove third-party credentials from the scheduler!");
         dbManager.removeThirdPartyCredential(ident.getUsername(), key);
     }
+    
+    @Override
+    public Page<TaskId> getTaskIds(String taskTag, long from, long to, boolean mytasks,
+            boolean running, boolean pending, boolean finished, int offset, int limit)
+                    throws NotConnectedException, PermissionException {
+        RestPageParameters params = new RestPageParameters(frontendState, "getTaskIds", from, to, mytasks,
+                running, pending, finished, offset, limit, taskTag);
+        Page<TaskInfo> pTaskInfo;
+        pTaskInfo = dbManager.getTasks(params.getFrom(), params.getTo(), params.getTag(),
+                params.getOffset(), params.getLimit(), params.getUserName(), params.isPending(), params.isRunning(),
+                params.isFinished(), null);
+        List<TaskId> lTaskId = new ArrayList<TaskId>(pTaskInfo.getList().size());
+        for (TaskInfo taskInfo : pTaskInfo.getList()) {
+            lTaskId.add(taskInfo.getTaskId());
+        }
+        return new Page<TaskId>(lTaskId, pTaskInfo.getSize());
+    }
+
+    @Override
+    public Page<TaskState> getTaskStates(String taskTag, long from, long to,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)